clearcontainers/runtime
# Release: 3.0.0-beta.2

# Release 3.0.0-beta.2

## Changes
- tests: increase run.go code coverage
- tests: increase code coverage in console.go
- tests: Only re-run unit-tests if not initially running as root
- scripts: Move release_notes.sh to scripts
- Update dev install guide
- Unit tests for exec.go
- .ci: Fix runtime CI scripts to work with Jenkins
- tests: Increase unit-test coverage on create.go
- tests: Run unit tests as root and non-root.
- tests: Increase oci.go unit-test coverage
- vendor: Update virtcontainers vendoring for updated StatusToOCIState
- Make run tests use cached bundle
- tests: add unit tests for state.go
- pause: Implement full unit test coverage
- kill: Implement full unit test coverage
- tests: add unit tests for run.go
- tests: Add tests for delete using mock framework
- tests: Increase unit-test coverage for list.go
- tests: Stop bundle creation timing out tests
- tests: Add unit tests for start.go
- Increase create test coverage
- tests: Rework OCI helper functions
- config: Add comment about ncpus >0 <=255
- state: Apply indentation to the state output.
- tests: Add helper functions
- test: add unit test for console Read and Write
- test: add unit test for isCgroupMounted function
- tests: switchable virtcontainers implementation
- vendor: Update virtcontainers vendoring for vcMock
- cc-check: warn if unrestricted_guest unavailable under hypervisor
- build: rebuild if git HEAD changed
- vendor: Update virtcontainers vendoring
- Docs: Add how to get agent debug logs.
- pause: Rewrite pause binary in go.
- docs: Remove references to alpha version
- docs: Update instruction to download compressed kernel
- docs : Automate release notes

## Shortlog
744f499 tests: increase run.go code coverage
b6ae359 tests: increase code coverage in console.go
fa25a03 tests: Only re-run unit-tests if not initially running as root
d17f710 scripts: Move release_notes.sh to scripts
b9f64a1 docs: Add more detail for gcc requirement
a96de62 docs: Remove requirement to have glibc-static installed
75014c0 tests: Add unit tests for exec.go
39c0a3b tests: Change newSingleContainerPodStatusList to include annotations
0cd9403 cli: Check for empty string for cli flag "process" in exec.go
5d56643 .ci: Fix runtime CI scripts to work with Jenkins
776ad37 tests: Increase unit-test coverage on create.go
4b3524e tests: Run unit tests as root and non-root.
4433de5 tests: Increase oci.go unit-test coverage
5bd0bce vendor: Update virtcontainers vendoring for updated StatusToOCIState
9ff688b tests: Make testRunContainerSetup() call makeOCIBundle()
200ef77 tests: Use specConfig rather than "config.json"
9df259b tests: Improve comments.
3e776d1 tests: add unit tests for state.go
540d517 pause: Implement full unit test coverage
4b38fc3 kill: Implement full unit test coverage
7af81a1 tests: add unit tests for run.go
1a8505d tests: Add tests for delete using mock framework
e487ab4 tests: Increase unit-test coverage for list.go
de7cb39 tests: Stop bundle creation timing out tests
b5b6085 tests: Add unit tests for start.go
d9413e7 tests: Increase unit-test coverage on create.go
2cac62a create: Remove redundant code.
b9a4d2f tests: Rework OCI helper functions
9b3b85f config: Add comment about ncpus >0 <=255
29e3d90 state: Apply indentation to json output from state cmd.
f2e6d08 tests: Add helper functions
2a5018a test: add unit test for console Read and Write
b4cd8fe tests: use testify to write unit tests
a143831 test: add unit test for isCgroupMounted function
1c7cbf4 tests: use testify in oci_test.go
15b879e tests: switchable virtcontainers implementation
9cfcc37 vendor: Update virtcontainers vendoring for vcMock
7e749ef cc-check/tests: Switch to using assert object.
ef5bb29 cc-check: warn if unrestricted_guest unavailable under hypervisor
16cab3d build: rebuild if git HEAD changes
839b484 vendor: Update virtcontainers vendoring
edf1261 Docs: Add how to get agent debug logs.
9f3ca75 pause: Rewrite pause binary in go.
93b127d docs: Remove references to alpha version
44edd61 docs: Update instruction to download compressed kernel
4674607 docs : Automate release notes

## Compatibility with Docker
Clear Containers 3.0.0-beta.2 is compatible with Docker v17.06-ce
## OCI Runtime Specification
Clear Containers 3.0.0-beta.2 support the OCI Runtime Specification [v1.0.0-rc5][ocispec]

## Clear Linux Containers image
Clear Containers 3.0.0-beta.2 requires at least Clear Linux containers image [16910][clearlinuximage]

## Clear Linux Containers Kernel
Clear Containers 3.0.0-beta.2 requires at least Clear Linux Containers  kernel [v4.9.35-76.container][kernel]

## Installation
- [Ubuntu][ubuntu]                                         
- [Fedora][fedora]                                         
- [Developers][developers]                                         


## Issues & limitations


More information [Limitations][limitations]

[clearlinuximage]: https://download.clearlinux.org/releases/16910/clear/clear-16910-containers.img.xz
[kernel]: https://github.com/clearcontainers/linux/tree/v4.9.35-76.container
[ocispec]: https://github.com/opencontainers/runtime-spec/releases/tag/v1.0.0-rc5
[limitations]: https://github.com/clearcontainers/runtime/blob/e84474ca45fecfef1200ec1c5b3008ea3643236d/docs/limitations.md
[ubuntu]: https://github.com/clearcontainers/runtime/blob/e84474ca45fecfef1200ec1c5b3008ea3643236d/docs/ubuntu-installation-guide.md
[fedora]: https://github.com/clearcontainers/runtime/blob/e84474ca45fecfef1200ec1c5b3008ea3643236d/docs/fedora-installation-guide.md
[developers]: https://github.com/clearcontainers/runtime/blob/e84474ca45fecfef1200ec1c5b3008ea3643236d/docs/developers-clear-containers-install.md
